kubernetes/kiali_monitoring/v1alpha1/kiali_monitoring_types.go

```diff
@@ -0,0 +1,72 @@
+package v1alpha1
+
+import (
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	// Raw constant for DataType
+	Raw = "raw"
+	// Rate constant for DataType
+	Rate = "rate"
+	// Histogram constant for DataType
+	Histogram = "histogram"
+)
+
+var KialiMonitoringGroupVersion = schema.GroupVersion{
+	Group:   "monitoring.kiali.io",
+	Version: "v1alpha1",
+}
+
+type MonitoringDashboard struct {
+	meta_v1.TypeMeta   `json:",inline"`
+	meta_v1.ObjectMeta `json:"metadata"`
+	Spec               MonitoringDashboardSpec `json:"spec"`
+}
+
+type MonitoringDashboardSpec struct {
+	Title   string                     `json:"title"`
+	Runtime string                     `json:"runtime"`
+	Charts  []MonitoringDashboardChart `json:"charts"`
+}
+
+type MonitoringDashboardChart struct {
+	Name         string                           `json:"name"`
+	Unit         string                           `json:"unit"`
+	Spans        int                              `json:"spans"`
+	MetricName   string                           `json:"metricName"`
+	DataType     string                           `json:"dataType"`   // DataType is either "raw", "rate" or "histogram"
+	Aggregator   string                           `json:"aggregator"` // Aggregator can be set for raw data. Ex: "sum", "avg". See https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators
+	Aggregations []MonitoringDashboardAggregation `json:"aggregations"`
+}
+
+type MonitoringDashboardAggregation struct {
+	Label       string `json:"label"`
+	DisplayName string `json:"displayName"`
+}
+
+// TODO: auto-generate the following deepcopy methods!
```

Comment:
@lucasponce I don't know what you think about that, maybe at some point we should add some code-generation tool to our build workflow to generate such code (that would also allow us to use kind-of generics!)

Reply:
+1 yeah, it's what I see most projects do.
Also, I saw that can be added as an "annotation" driven by the go generate phase.
(It's what I saw also in Istio while tackling with the adapters code).